### DISCUSSION TITLE
pydantic objects stop validating variable changes after instantiation?

### DISCUSSION THREAD

**Discussion by ryan-minato** [pydantic/pydantic#5160] [Question] [answered]
I initially assumed that pydantic would continue to perform type checks and implicit type conversion when variables are modified after instantiation. 
However, I noticed that when I changed a variable to an invalid type, it did not raise an error and the variable was changed to an incorrect type.

Please see the following code as an example:
``` python
>>> from pydantic import BaseModel

>>> class SampleType(BaseModel):
>>>     a: int
>>>     b: str

>>> SampleType(a=1, b='aaa')
ValidationError: 1 validation error for SampleType
a
  value is not a valid integer (type=type_error.integer)

>>> t = SampleType(a=1, b='aaa')
>>> t
SampleType(a='bbb', b='aaa')

>>> t.a = 'bbb'
>>> t
SampleType(a='bbb', b='aaa')
```

**I would like to ask if this behavior is expected or if it is a bug.** 

I have tested this behavior in different environments, including the Colab environment, and found that it is consistent across all environments. 
Is this behavior expected, or is it a bug?

**Comment by nymous** [comment, marked as answer]
Hi!
Yes this behavior is expected, you can configure it with `validate_assignment = True`: https://docs.pydantic.dev/usage/model_config/

**Reply by samuelcolvin** [reply]
Thanks for answering.
